projects/client/RabbitMQ.Client/src/client/impl/ConsumerWorkService.cs

```diff
@@ -49,46 +40,34 @@ public void StopWork()
         class WorkPool
         {
             readonly ConcurrentQueue<Action> actions;
-            readonly AutoResetEvent messageArrived;
-            readonly TimeSpan waitTime;
+            readonly SemaphoreSlim semaphore = new SemaphoreSlim(0);
             readonly CancellationTokenSource tokenSource;
-            readonly string name;
+            private Task worker;
 
             public WorkPool(IModel model)
             {
-                name = model.ToString();
                 actions = new ConcurrentQueue<Action>();
-                messageArrived = new AutoResetEvent(false);
-                waitTime = TimeSpan.FromMilliseconds(100);
                 tokenSource = new CancellationTokenSource();
             }
 
             public void Start()
             {
-#if NETFX_CORE
-                System.Threading.Tasks.Task.Factory.StartNew(Loop, System.Threading.Tasks.TaskCreationOptions.LongRunning);
-#else
-                var thread = new Thread(Loop)
-                {
-                    Name = "WorkPool-" + name,
-                    IsBackground = true
-                };
-                thread.Start();
-#endif
+                worker = Task.Run(Loop);
```

Comment:
Use ThreadPool rather than a Thread. Turns the Loop into an async Task instead, for better utilization and gets rid of polling mechanism.